client/packages/system/src/Patient/DocumentHistory/DocumentHistory.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Typography,
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+} from '@mui/material';
+import { ChevronDownIcon } from '@common/icons';
+import { DocumentNode } from 'packages/common/src/types/schema';
+import React, { FC, useEffect, useState } from 'react';
+import ReactDiffViewer from 'react-diff-viewer';
+import { usePatient } from '../api';
+import { Box } from 'packages/common/src';
+
+export const DocumentHistory: FC<{ documentName: string }> = ({
+  documentName,
+}) => {
+  const [history, setHistory] = useState([] as DocumentNode[]);
```

Comment:
I don't see the need for local state here.

I would remove the `useEffect` and change the hook call to be 
```  
const { data: docHistory, isLoading } = usePatient.document.history(documentName);
```

and add

```
  if (isLoading) return <BasicSpinner />;

```

and `(docHistory ?? []).map((h, i)` to make typescript happier ( also change the diff viewer usage ).

Reply:
thanks done, I just check if the return value is not undefined...